src/coreclr/src/System.Private.CoreLib/src/System/Math.CoreCLR.cs

```diff
@@ -11,119 +11,176 @@
 ===========================================================*/
 
 using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
 
 namespace System
 {
     public static partial class Math
     {
+        [SuppressGCTransition]
+        [DllImport(RuntimeHelpers.QCall, CallingConvention = CallingConvention.Cdecl)]
+        private static extern double AbsInternal(double value);
         [Intrinsic]
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern double Abs(double value);
+        public static double Abs(double value) => AbsInternal(value);
```

Comment:
What is the reason for having the public method call an internal QCall, rather than just having the public method itself be a QCall?

I saw Jan's comment here (https://github.com/dotnet/runtime/pull/39474#discussion_r457466630), but it isn't clear to me why we want to avoid exporting the main entry point. It just seems like more JIT/inlining work and I would guess would result in worse codegen when optimizations are disabled or we are on tier 0.

Reply:
An assert in ecall.cpp was failing: https://github.com/dotnet/runtime/pull/39474#discussion_r457456002.

Reply:
> What is the reason for having the public method call an internal QCall, rather than just having the public method itself be a QCall?

IIRC, the `src/libraries` tooling does not like exporting DllImports. Also, I was not sure how the JIT will dead with `DllImports` that are intrinsics at the same time.

I was assuming that it will inline just fine, but it is not the case unfortunately. I agree that it may be better to export DllImports directly and fix the problems that it is going to hit instead.

Reply:
> An assert in ecall.cpp was failing: https://github.com/dotnet/runtime/pull/39474#discussion_r457456002.
> I agree that it may be better to export DllImports directly and fix the problems that it is going to hit instead.

The assert does in ecall.cpp does not like `QCall` methods to be public.

Reply:
> An assert in ecall.cpp was failing: https://github.com/dotnet/runtime/pull/39474#discussion_r457456002.
> I agree that it may be better to export DllImports directly and fix the problems that it is going to hit instead.

The assert in ecall.cpp does not like `QCall` methods to be public.